tests/FakeItEasy.Analyzer.CSharp.Tests/ArgumentConstraintOutsideCallSpecAnalyzerTests.cs

```diff
@@ -0,0 +1,224 @@
+﻿namespace FakeItEasy.Analyzer.CSharp.Tests
+{
+    using System.Collections.Generic;
+    using FakeItEasy.Analyzer.Tests.Helpers;
+    using FakeItEasy.Tests;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Xunit;
+
+    public class ArgumentConstraintOutsideCallSpecAnalyzerTests : DiagnosticVerifier
+    {
+        public static IEnumerable<object[]> Constraints =>
+            TestCases.FromObject(
+                "A<int>._",
+                "A<int>.Ignored",
+                "A<int>.That.IsEqualTo(42)",
+                "A<int>.That.Not.IsEqualTo(42)");
+
+        [Theory]
+        [InlineData("A<int>._")]
+        [InlineData("A<int>.Ignored")]
+        [InlineData("A<int>.That")]
+        [InlineData("A<int>.That.Not")]
+        [InlineData("A<int>.That.IsEqualTo(42)")]
+        [InlineData("A<int>.That.Not.IsEqualTo(42)")]

```

Comment:
I probably wouldn't have thought to add the extra cases (`That`, `That.Not`). Nice.